gobblin-runtime/src/main/java/gobblin/runtime/Fork.java

```diff
@@ -403,18 +419,33 @@ private void processRecords()
     while (true) {
       try {
         Object record = this.recordQueue.get();
-        if (record == null) {
+        if (record == null || record == SHUTDOWN_RECORD) {
           // The parent task has already done pulling records so no new record means this fork is done
           if (this.parentTaskDone) {
             return;
+          } else {
+            this.logger.error("Found a {} record but parent task is not done. Aborting this fork.", record);
```

Comment:
as per what I see .. it should never get in else clause? did I miss some condition?